flake.nix

```diff
@@ -0,0 +1,149 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs?rev=ea5234e7073d5f44728c499192544a84244bf35a";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachSystem [ "x86_64-linux" "x86_64-windows" "x86_64-darwin" "aarch64-darwin" ] (targetSystem:
+      let
+        # This needs to be made dynamic using something like builtins.currentSystem
+        buildSystem = "x86_64-linux";
+
+        systems = {
+          "x86_64-linux" = [ "linux" "x64" ];
+          "x86_64-windows" = [ "win32" "x64" ];
+          "x86_64-darwin" = [ "darwin" "x64" ];
+          "aarch64-darwin" = [ "darwin" "arm64" ];
+        };
+
+        platform = builtins.elemAt systems.${targetSystem} 0;
+        arch = builtins.elemAt systems.${targetSystem} 1;
+
+        pkgs = import nixpkgs {
+          system = buildSystem;
+        };
+
+        utils = pkgs.callPackage ./utils.nix {};
+
+        commitHash = toString (self.rev or self.dirtyRev);
+        npmDepsHash = builtins.readFile ./npmDepsHash;
+
+        polykey_cli-src = pkgs.buildNpmPackage {
+          inherit npmDepsHash;
+          pname = utils.packageName;
+          version = utils.packageVersion;
+          src = utils.src;
+          COMMIT_HASH = commitHash;
+          GIT_DIR = if commitHash != null then null else utils.dotGit;
+          postInstall = ''
+            mv "$packageOut"/build/build.json "$out"/build.json;
+            rm -rf \
+              "$packageOut"/build \
+              "$packageOut"/src \
+              "$packageOut"/.env.example \
+              "$packageOut"/images \
+              "$packageOut"/scripts \
+              "$packageOut"/tsconfig.build.json \
+              "$packageOut"/tsconfig.json \
+              "$packageOut"/LICENSE \
+              "$packageOut"/ADDITIONAL_TERMS \
+              "$packageOut"/README.md;
+          '';
+        };
+
+        polykey_cli-release = pkgs.buildNpmPackage {
```

Comment:
Do not use `x_a-b` style naming, too complicated and too prone to error. We only ever use `a-b` naming for final attributes in nix.

Reply:
Is it standard practice to name it the name of the application, or can flakes also just have a "default" attribute?

Reply:
What you see there, `polykey_cli-release` is just a variable assigned to the derivation produced from `buildNpmPackage`. That name does not change anything. What does change things typically in a flake, is in the `packages` section in the flake outputs, the name of the package will change the binary name. In the case of the `vercel/pkg` system, since that uses a custom output system it will always output under the binary `result`. Same for the docker image.

Reply:
Also the naming is fixed in 1d7af25.